jwst/outlier_detection/outlier_detection.py

```diff
@@ -347,6 +321,83 @@ def detect_outliers(self, blot_models):
             for i in range(len(self.input_models)):
                 self.inputs.dq[i, :, :] = self.input_models[i].dq
 
+    def driz_weight_no_resample(self):
+        """
+        give weights to drizzled model without resampling
+        """
+        drizzled_models = self.input_models
+        for i in range(len(self.input_models)):
+            drizzled_models[i].wht = build_driz_weight(
+                self.input_models[i],
+                weight_type=self.outlierpars['weight_type'],
+                good_bits=self.outlierpars['good_bits'])
+        return drizzled_models
+
+
+class OutlierDetectionImaging(OutlierDetection):
+    """sub-class for performing outlier detection on JWST imaging observations."""
+
+    def __init__(self, input_models, **pars):
+        
+        super().__init__(input_models, **pars)
+    
+
+    def do_detection(self):
+        """Flag outlier pixels in DQ of input images."""
+        pars = self.outlierpars
+
+        if pars['resample_data']:
+            # Start by creating resampled/mosaic images for
+            # each group of exposures
+            resamp = resample.ResampleData(self.input_models, single=True,
+                                           blendheaders=False, **pars)
+            drizzled_models = resamp.do_drizzle()
+
+        else:
+            # for non-dithered data, the resampled image is just the original image
+            drizzled_models = self.driz_weight_no_resample()
+        
+        # compute median
+        median_data = self.create_median(drizzled_models)
+
+        # Initialize intermediate products used in the outlier detection
+        with datamodel_open(drizzled_models[0]) as dm0:
+            median_model = datamodels.ImageModel(dm0.data.shape)
+            median_model.update(dm0)
+            median_model.meta.wcs = dm0.meta.wcs
+
+        # Perform median combination on set of drizzled mosaics
+        median_model.data = median_data[0] #all planes are the same
+        if pars['save_intermediate_results']:
+            if self.outlierpars.get('asn_id', None) is None:
+                suffix_to_remove = self.resample_suffix
+            else:
+                suffix_to_remove = f"_{self.outlierpars['asn_id']}{self.resample_suffix}"
+            median_model_output_path = self.make_output_path(
+                basepath=median_model.meta.filename.replace(suffix_to_remove, '.fits'),
+                suffix='median')
+            median_model.save(median_model_output_path)
+            log.info(f"Saved model in {median_model_output_path}")
+
+        if pars['resample_data']:
+            # Blot the median image back to recreate each input image specified
+            # in the original input list/ASN/ModelContainer
+            blot_models = self.blot_median(median_model)
+
+        else:
+            # Median image will serve as blot image
+            blot_models = ModelContainer(open_models=False)
```

Comment:
`open_models` is used even though it is not an understood keyword argument for `ModelContainer`. It doesn't produce an error (and the existing step uses this unknown argument) but it also has no effect. If the goal is to have the container work on filenames instead of models some combination of `save_open` and `return_open` could be set.